channeldb/graph.go

```diff
@@ -491,9 +502,20 @@ func (c *ChannelGraph) deleteLightningNode(nodes *bbolt.Bucket,
 // the channel supports. The chanPoint and chanID are used to uniquely identify
 // the edge globally within the database.
 func (c *ChannelGraph) AddChannelEdge(edge *ChannelEdgeInfo) error {
-	return c.db.Batch(func(tx *bbolt.Tx) error {
+	c.cacheMu.Lock()
+	defer c.cacheMu.Unlock()
+
+	err := c.db.Update(func(tx *bbolt.Tx) error {
 		return c.addChannelEdge(tx, edge)
 	})
+	if err != nil {
+		return err
+	}
+
+	c.rejectCache.remove(edge.ChannelID)
+	c.chanCache.remove(edge.ChannelID)
```

Comment:
Since we update the cache now when we add/update edge policies, we may need to also add the edge info to the cache here. Otherwise, it seems possible that we add two edges for a `ChannelEdge`, but then never update the edge info as well, causing us to send a blank edge info.

Reply:
Scratch that: items are only added for the first time within `ChanUpdatesInHorizon`, so when we add the edge policy, only if it's already present will we update the edge policies in the cache.

Reply:
yep exactly!